### PR TITLE
i#3544, i#1551, i#1569 signal: Remove a non-existing field in sigframe_rt_t

### PR DESCRIPTION
**Pull request by ksco**
The `retcode` field in `sigframe_rt_t` is only available in x86 and AArch32, this fixes a buffer overflow in `memcpy_rt_frame()` on RISC-V.

Issue: #3544 #1551 #1569

### CODE REVIEW

File: core/unix/signal_private.h [hunk 1]
```diff
@@ -280,11 +284,9 @@ typedef struct rt_sigframe {
 #    elif defined(AARCHXX)
     kernel_siginfo_t info;
     kernel_ucontext_t uc;
-    char retcode[RETCODE_SIZE];
```
Comment (derekbruening) [inline core/unix/signal_private.h:283]: I believe AArch32 has retcode in the frame set up by the kernel.  I think it is not just x86: also AArch32.

File: core/unix/signal.c [hunk 2]
```diff
@@ -3644,7 +3644,6 @@ convert_frame_to_nonrt(dcontext_t *dcontext, int sig, sigframe_rt_t *f_old,
     f_new->sig_noclobber = f_new->sig;
 #    elif defined(ARM)
     memcpy(&f_new->uc, &f_old->uc, sizeof(f_new->uc));
-    memcpy(f_new->retcode, f_old->retcode, sizeof(f_new->retcode));
```
Comment (derekbruening) [inline core/unix/signal.c:3647]: I think AArch32 does have retcode: I see it in https://elixir.bootlin.com/linux/latest/source/arch/arm/kernel/signal.c#L402

File: core/unix/signal_private.h [hunk 3]
```diff
@@ -214,7 +214,11 @@ typedef _STRUCT_UCONTEXT /* == __darwin_ucontext */ kernel_ucontext_t;
  * (these are from /usr/src/linux/arch/i386/kernel/signal.c for kernel 2.4.17)
  */
 
-#    define RETCODE_SIZE 8
+#    if defined(X86)
+#        define RETCODE_SIZE 8
+#    elif defined(ARM)
+#        define RETCODE_SIZE 16
```
Comment (derekbruening) [inline core/unix/signal_private.h:220]: Hmm, could you explain where this came from?

File: core/unix/signal.c (hunk 2 above)
Comment (ksco) [reply core/unix/signal.c]: I'm very sorry for misreading this, yes, AArch32 has a retcode.

File: core/unix/signal_private.h (hunk 3 above)
Comment (ksco) [reply core/unix/signal_private.h:220]: It's from https://elixir.bootlin.com/linux/latest/source/arch/arm/kernel/signal.h#L5

File: core/unix/signal_private.h (hunk 1 above)
Comment (ksco) [reply core/unix/signal_private.h:283]: Yes, you're right, there is a `retcode` in AArch32, but not in AArch64.

File: core/unix/signal_private.h (hunk 3 above)
Comment (derekbruening) [reply core/unix/signal_private.h:220]: I see, thanks.

File: core/unix/signal_private.h [hunk 4]
```diff
@@ -214,7 +214,11 @@ typedef _STRUCT_UCONTEXT /* == __darwin_ucontext */ kernel_ucontext_t;
  * (these are from /usr/src/linux/arch/i386/kernel/signal.c for kernel 2.4.17)
```
Comment (derekbruening) [inline core/unix/signal_private.h:214]: Please update the PR/merge description to include arm.

**Review by derekbruening — APPROVED** [review]

**Pull request merged**